appscale/tools/agents/ec2_agent.py

```diff
@@ -796,12 +908,19 @@ def cleanup_state(self, parameters):
 
     AppScaleLogger.log("Deleting security group {0}".format(
       parameters[self.PARAM_GROUP]))
-    while True:
+    retries_left = self.SECURITY_GROUP_RETRY_COUNT
+    while retries_left:
```

Comment:
If an EC2ResponseError is raised more than the number of retries, this function will return successfully.

Reply:
would:
```
        if retries_left == self.SECURITY_GROUP_RETRY_COUNT:
          raise
```

in the except clause be sufficient you think?

Reply:
That should be fine. However, since you're counting down, you'll probably want to compare to 0 rather than the retry count. You'll also likely want to change to `while True` since you are moving your exit condition.